IdentityCore/src/oauth2/aad_v1/MSIDAADV1Oauth2Factory.m

```diff
@@ -162,4 +163,55 @@ - (MSIDLegacySingleResourceToken *)legacyTokenFromResponse:(MSIDTokenResponse *)
     return nil;
 }
 
+- (NSURL *)generateStartURL:(MSIDRequestParameters *)requestParams
+{
+    NSString* state = [self encodeProtocolState:requestParams];
+    
+    // if value is nil, it won't appear in the dictionary
+    NSMutableDictionary *queryParams = [NSMutableDictionary dictionaryWithObjectsAndKeys:
```

Comment:
nit: we can also use URLQueryItem and URLComponents

Reply:
Sure, let me try that :)

Reply:
Actually, you can leave this part, 
I think it might be better to change the msidURLFormEncode method to use queryitem over string manipulation

Reply:
Refer to https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/126/commits/c669afefd303cd28f1c8a09c86954f06fe58edaa

this way, we can create NSURLComponent, and set scheme, host, path, queryitems which might seem a little more elegant

Reply:
Since we may need to see if V1 and V2 have something in common for constructing the start URL, so I'd better make the change later :)